docs/CMakeLists.txt

```diff
@@ -28,406 +28,51 @@
  #################################################################################
 
 include(CMakeParseArguments)
-
-if(${MSVC})
-    option(IVW_DOXYGEN_PROJECT "Create Inviwo doxygen files" OFF)
-    if(${IVW_DOXYGEN_PROJECT})
-        message(STATUS
-            "In Visual Studio the doxygen project will rerun every time you run\n"
-            "\"ALL_BUILD\" even if it is up to date. Hence, you propbably want\n"
-            "to disable generation of doxygen on each build this can be done\n"
-            "by right clicking on the docs folder in the \"Solution Explorer\" and\n"
-            "select \"Unload Project\".")
-    endif()
+include(CMakeDependentOption)
+
+option(IVW_DOXYGEN_PROJECT "Create Inviwo doxygen files" OFF)
+cmake_dependent_option(IVW_DOXYGEN_OPEN_HTML_AFTER_BUILD
+                       "Open the generated doxygen HTML when build is done"
+                       OFF
+                       "IVW_DOXYGEN_PROJECT"
+                       OFF)
+
+if(IVW_DOXYGEN_PROJECT)
+    # sets: DOXYGEN_FOUND, DOXYGEN_EXECUTABLE, DOXYGEN_DOT_FOUND, DOXYGEN_DOT_EXECUTABLE
+    find_package(Doxygen REQUIRED)
+    find_package(Python3 COMPONENTS Interpreter REQUIRED)
```

Comment:
Can sphinx be added here?

Reply:
at the moment I use python3 -m sphinx ... to call it. There is also a script sphinx-build that one can use. But it seems safer to use the module directly. Maybe it would be possible to add some check to see it the sphinx module is installed. And you will only run into that issue when building the target. On the other hand you can then also just pip install it and don't have to rerun cmake...

Reply:
Is it possible to do add it like we do with numpy:
find_package(Python3 COMPONENTS Interpreter Development **NumPy** REQUIRED)

Reply:
That would have been nice, But I think numpy is a special case https://cmake.org/cmake/help/latest/module/FindPython3.html